app/src/main/java/ch/admin/bag/dp3t/checkin/diary/EditDiaryEntryFragment.java

```diff
@@ -0,0 +1,144 @@
+		return fragment;
+	}
+
+	@Override
+	public void onCreate(@Nullable Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		diaryStorage = DiaryStorage.getInstance(requireContext());
+		long diaryEntryId = getArguments().getLong(ARG_DIARY_ENTRY_ID);
+		diaryEntry = diaryStorage.getDiaryEntryWithId(diaryEntryId);
+	}
+
+	@Override
+	public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+		titleTextView = view.findViewById(R.id.check_out_fragment_title);
+		subtitleTextView = view.findViewById(R.id.check_out_fragment_subtitle);
+		doneButton = view.findViewById(R.id.check_out_fragment_done_button);
+		cancelButton = view.findViewById(R.id.check_out_fragment_cancel_button);
+		commentEditText = view.findViewById(R.id.check_out_fragment_comment_edit_text);
+		fromTime = view.findViewById(R.id.check_out_fragment_from_text_view);
+		toTime = view.findViewById(R.id.check_out_fragment_to_text_view);
+		dateTextView = view.findViewById(R.id.check_out_fragment_date);
+		hideInDiaryButton = view.findViewById(R.id.edit_diary_entry_hide_from_diary_button);
+
+		titleTextView.setText(diaryEntry.getVenueInfo().getTitle());
+		subtitleTextView.setText(VenueInfoExtensionsKt.getSubtitle(diaryEntry.getVenueInfo()));
+
+		refreshTimeTextViews();
+
+		fromTime.setOnClickListener(v -> showTimePicker(true));
+		toTime.setOnClickListener(v -> showTimePicker(false));
+		commentEditText.setText(diaryEntry.getComment());
+
+		hideInDiaryButton.setVisibility(View.VISIBLE);
+		hideInDiaryButton.setOnClickListener(v -> hideInDiary());
+
+		doneButton.setOnClickListener(v -> {
+			saveEntry();
+			requireActivity().getSupportFragmentManager().popBackStack();
+		});
+		cancelButton.setOnClickListener(v -> {
+			requireActivity().getSupportFragmentManager().popBackStack();
+		});
+	}
+
+	private void refreshTimeTextViews() {
+		fromTime.setText(StringUtil.getHourMinuteTimeString(diaryEntry.getArrivalTime(), "  :  "));
+		toTime.setText(StringUtil.getHourMinuteTimeString(diaryEntry.getDepartureTime(), "  :  "));
+		dateTextView.setText(StringUtil.getCheckOutDateString(getContext(), diaryEntry.getArrivalTime(),
+				diaryEntry.getDepartureTime()));
+	}
+
+	private void showTimePicker(boolean isFromTime) {
+		Calendar time = Calendar.getInstance();
+		if (isFromTime) {
+			time.setTimeInMillis(diaryEntry.getArrivalTime());
+		} else {
+			time.setTimeInMillis(diaryEntry.getDepartureTime());
+		}
+		int hour = time.get(Calendar.HOUR_OF_DAY);
+		int minute = time.get(Calendar.MINUTE);
+		TimePickerDialog timePicker;
+		timePicker = new TimePickerDialog(getContext(), (picker, selectedHour, selectedMinute) -> {
+			time.set(Calendar.HOUR_OF_DAY, selectedHour);
+			time.set(Calendar.MINUTE, selectedMinute);
+			if (isFromTime) {
+				diaryEntry.setArrivalTime(time.getTimeInMillis());
+			} else {
+				diaryEntry.setDepartureTime(time.getTimeInMillis());
+			}
+			if (diaryEntry.getDepartureTime() < diaryEntry.getArrivalTime()) {
```

Comment:
I think we can remove this logic since we do not restrict check ins to a maximum of 24 hours anymore, or do we? 

If we do not, we also need a DateTimePicker instead of a TimePicker Dialog. 

And I think we should enforce non-overlapping check ins.